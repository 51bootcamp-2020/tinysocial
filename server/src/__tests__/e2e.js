// E2E test
'use strict';
require('dotenv').config();
const {ApolloServer} = require('apollo-server-express');
const typeDefs = require('../schema');
const resolvers = require('../resolvers');

const {EventAPI} = require('../dataSources/eventAPI');
const {ReviewAPI} = require('../dataSources/reviewAPI');
const {AuthAPI} = require('../dataSources/authAPI');
const {TagAPI} = require('../dataSources/tagAPI');
const {UserAPI} = require('../dataSources/userAPI.js');
const {createStore} = require('../database');
const context = require('../context');

const gql = require('graphql-tag');

const {startTestServer, toPromise} = require('./__utils');

const EVENT_LIST_REQUEST_QUERY = gql`
    query ($pageSize: Int){
        events(pageSize: $pageSize){
            cursor,
            events{
                id,
                title,
                description
            }
        }
    }`;

const EVENT_DETAIL_REQUEST_QUERY = gql`
    query getEvent($eventId: ID!) {
        event(id: $eventId) {
            id,
            title,
            thumbnailUrl,
            description,
            price,
            tags {
                id
                name
            }
            ... on EventBookClub {
                bookTitle
                bookDescription
                bookAuthor
            }
            schedule {
                id
                startDateTime
                endDateTime
                address
                latitude
                longitude
            }
            host {
                firstName,
                lastName,
                selfDescription,
                profileImgUrl
            }
        }
    }`;

describe('Server - e2e', () => {
  let stop;
  let graphql;

  beforeEach(async () => {
    const store = await createStore();
    const dataSources = () => ({
      eventAPI: new EventAPI(store),
      reviewAPI: new ReviewAPI(store),
      tagAPI: new TagAPI(store),
      userAPI: new UserAPI(store),
      authAPI: new AuthAPI(store),
    });

    const server = new ApolloServer({
      typeDefs,
      resolvers,
      dataSources,
      context,
    });
    const testServer = await startTestServer(server);
    stop = testServer.stop;
    graphql = testServer.graphql;
  });

  afterEach(() => stop());

  it('gets list of events', async () => {
    const res = await toPromise(
        graphql({
          query: EVENT_LIST_REQUEST_QUERY,
          variables: {pageSize: 9},
        }),
    );

    expect(res).toMatchSnapshot();
  });

  it('gets a single event', async () => {
    const res = await toPromise(
        graphql({query: EVENT_DETAIL_REQUEST_QUERY, variables: {eventId: 2}}),
    );

    expect(res).toMatchSnapshot();
  });
});
