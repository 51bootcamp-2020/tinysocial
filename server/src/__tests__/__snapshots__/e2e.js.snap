// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e gets a single event 1`] = `
Object {
  "data": Object {
    "event": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "locations": Array [
        Object {
          "column": 3,
          "line": 2,
        },
      ],
      "message": "Data cannot be resolved to any Event type. Please contact to backend developers.",
      "path": Array [
        "event",
      ],
    },
  ],
}
`;

exports[`Server - e2e gets list of events 1`] = `
Object {
  "data": Object {
    "events": Object {
      "cursor": 0,
      "events": Array [],
    },
  },
}
`;
